### PR TITLE
Distance Calculator demo with helm

### PR DESCRIPTION
**Pull request by Sanjay1611**
This PR does following

- bump flink and corresponding libraries and connectors to latest version (flink 1.15.2)
- modify `mqttwriter` docker file for packaging with maven (instead of gradle)
- adding an entrypoint script for `mqttwriter` docker file to run mqttwriter jar
- automating the editing of Distance.csv in the entrypoint script
- adds following helm charts for distance calculator demo
  - features - This will deploy a custom ProjectFeature `PravegaMqttBroker` (enhanced for having the credentials)
  - calculator - This will deploy a project with needed projectfeatures, flinkcluster and distance calculator flinkapplication
  - mqttwriter - This will run mqttwriter jar in a Pod with corresponding envs set

Order of running helmcharts will be like below
- helm install features
- helm install calculator
- helm install mqttwriter

Custom docker build of mqttwriter app can be triggered by 
- `cd distance-calculator/mqttwriter/`
- `docker build -t <image_name> .

To run the custom image of mqttwriter 
- `cd distance-calculator/charts/`
- `vi values.yaml`
- edit `mqtt.image`

Note - This is a dependent PR on https://github.com/vangork/sdp-starter-kit/pull/31

### CODE REVIEW

**Comment by Sanjay1611** [comment]
MicrosoftTeams-image

File: distance-calculator/charts/features/templates/broker_feature.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+apiVersion: nautilus.dellemc.com/v1alpha1
+kind: ProjectFeature
```
Comment (Vi-Nk) [inline distance-calculator/charts/features/templates/broker_feature.yaml:2]: @Bilna if this chart is capable of deploying project feature on its own , do we need to remove the step in documents of doing the same in UI ? 
Or do we say that whatever you are trying to do in UI, can be replaced with deployable charts and provide it as a separate document section either in repo README or in code-hub doc

File: distance-calculator/charts/mqttwriter/templates/mqttwriter.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{{- $mqtt := .Values.mqtt -}}
+apiVersion: v1
+kind: Pod
```
Comment (Vi-Nk) [inline distance-calculator/charts/mqttwriter/templates/mqttwriter.yaml:3]: @Sanjay1611 , @Bilna are we planning on deploying mqtt writer as a pod too ? or do we use it as a client / binary outside SDP to push simulated/real data from local file. 
I understand that both of these are feasible solutions , but we might need to be clear on what direction the deliverable has to be for helm charts. (All deployable modules / deployable helm modules( mqtt broker , flink cluster,app , grafana ) +external ingestion clients (mqtt writer)

File: distance-calculator/charts/features/templates/broker_feature.yaml (hunk 1 above)
Comment (Bilna) [reply distance-calculator/charts/features/templates/broker_feature.yaml:2]: We can demo it in two ways. 
1. using the Helm 
3. directly using SDP UI

So, the documentation we can provide. Users can choose which way they want to install

File: distance-calculator/charts/mqttwriter/templates/mqttwriter.yaml (hunk 2 above)
Comment (Bilna) [reply distance-calculator/charts/mqttwriter/templates/mqttwriter.yaml:3]: We can deploy mqtt-writer as a pod

Comment (Sanjay1611) [reply distance-calculator/charts/mqttwriter/templates/mqttwriter.yaml:3]: @Vi-Nk Thats what was not clear in the story i.e. upto what extent automation has to be provided. But we did not hand to engage into a discussion. So I think for now running it as a pod should be fine ... we can modify the charts as per the requirement

**Review by vangork — COMMENTED** [review]
can add the grafanadashboard in the helm chart template for server monitor also.

**Comment by Sanjay1611** [comment]
> can add the grafanadashboard in the helm chart template for server monitor also.

Yes but we need to edit the datasource name in the grafana dashboard manually in the dashboard's json file

**Comment by Vi-Nk** [comment]
> > can add the grafanadashboard in the helm chart template for server monitor also.
> 
> Yes but we need to edit the datasource name in the grafana dashboard manually in the dashboard's json file. And we will not know its name since its generated dynamically

As we are using Flink Cluster metrics Influx as source , the name of datasource seem to be getting generated as <Flink_Cluster Name>_flnk . So If we are deploying flink cluster as a chart with predetermined name , the datasource name should also be predictable is what I feel.

**Comment by Sanjay1611** [comment]
Dashboard has been added as a part of `calculator` chart. Now we can directly view the calculator dashboard without importing it manually.

One more deployer script has been added so as to reduce the manual steps. It can be used as follows -
`./deploy.sh <name_of_the_application>`

**Comment by Bilna** [comment]
LGTM!

**Review by Bilna — APPROVED** [review]

**Pull request merged**